File: LeetCode/Two_Sum/Two_Sum.cpp
// Time Complexty: O(n) where n is the size of nums
// Space Complexty: O(n) where n is the size of nums

#include <unordered_map>

class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        std::unordered_map<int, int> map;
        vector<int> pair;
        
        for (int i=0; i<nums.size(); i++) {
            int value = nums[i];
            if (map.find(value) == map.end()) {
                map[target-value] = i;
            } else {
                pair.push_back(map[value]);
                pair.push_back(i);
                return pair;
            }
        }
        return pair;
    }
};

File: LeetCode/Two_Sum/Two_Sum.py
# Time Complexty: Linear, O(n) where n is the length of nums
# Space Complexty: Linear, 0(n) where n in the length of nums

class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        compliments = {}

        for i in range(len(nums)):
            if target-nums[i] in compliments:
                return [compliments[target-nums[i]],i]
            else:
                compliments[nums[i]] = i
